controller/src/main/java/io/pravega/controller/store/stream/InMemoryStream.java

```diff
@@ -959,6 +966,79 @@ private void handleStreamMetadataExists(final long timestamp, CompletableFuture<
         return CompletableFuture.completedFuture(null);
     }
 
+    @Override
+    CompletableFuture<Void> createWriterMarkRecord(String writer, long timestamp, ImmutableMap<Long, Long> position) {
+        CompletableFuture<Void> result = new CompletableFuture<>();
+        WriterMark mark = new WriterMark(timestamp, position);
+
+        synchronized (writersLock) {
+            VersionedMetadata<WriterMark> existing = writerMarks.get(writer);
+            if (existing != null) {
+                result.completeExceptionally(StoreException.create(StoreException.Type.DATA_EXISTS, "writer mark exists"));
+            } else {
+                writerMarks.put(writer, new VersionedMetadata<>(mark, new Version.IntVersion(0)));
+                result.complete(null);
+            }
+        }
+        
+        return result;
+    }
+
+    @Override
+    public CompletableFuture<Void> removeWriter(String writer) {
+        CompletableFuture<Void> result = new CompletableFuture<>();
```

Comment:
Here and in the other methods in this file, you can use `CompletableFuture.completedFuture(null)`